apps/platform/src/storage/S3StorageProvider.ts

```diff
@@ -47,4 +51,11 @@ export class S3StorageProvider implements StorageProvider {
             Key: filename,
         })
     }
+
+    static url(path: string) {
+        const config = App.main.env.storage as S3Config
+        return config.endpoint && config.forcePathStyle
+            ? combineURLs([config.endpoint, config.bucket, path])
+            : combineURLs([`https://${config.bucket}.s3.amazonaws.com`, path])
```

Comment:
You can not pass endpoint to config, in case it is AWS. It is resolved by SDK by default. Just FYI.